bot.py

```diff
@@ -1,76 +1,61 @@
-      "..######...######...#######..########...#######.....##...")
-print("----------------------------------------------------------")
-print("v" + bot.version)
+import time
+
+
+PREFIX = "?"
+DESCRIPTION = "scubot"
+BOT_VERSION = "2.0.0"
+CONFIG_PATH = "config.json"
+TOKEN_PATH = "token.json"
+
+
+try:
+    with open(CONFIG_PATH, 'r') as fp:
+        config = json.load(fp)
+        modules_to_load = config["load_modules"]
+except FileNotFoundError:
+    print("[FATAL] No config.json file found. Startup aborted.")
+    sys.exit()
+
+try:
+    with open(TOKEN_PATH, 'r') as fp:
+        token = json.load(fp)["token"]
+except FileNotFoundError:
+    print("[FATAL] No token file found. Startup aborted.")
+    sys.exit()
+
+print("""
+----------------------------------------------------------
+..######...######..##.....##.########...#######..########
+.##....##.##....##.##.....##.##.....##.##.....##....##...
+.##.......##.......##.....##.##.....##.##.....##....##...
+..######..##.......##.....##.########..##.....##....##...
+.......##.##.......##.....##.##.....##.##.....##....##...
+.##....##.##....##.##.....##.##.....##.##.....##....##...
+..######...######...#######..########...#######.....##...
+----------------------------------------------------------
+""")
+print(f"v{BOT_VERSION}")
 print("Logging in...")
 
 
-@bot.event
-async def on_ready():
-    print('Bot online as ' + str(bot.user))
-
-
-@bot.event
-async def on_command_error(ctx, error):
-    if isinstance(error, discord.ext.commands.CommandNotFound):  # Command not found, not our fault.
-        return
-    if isinstance(error, discord.ext.commands.MissingRequiredArgument):  # Missing some sort of argument - who knows?
-        await ctx.send("[!] Missing required argument.")
-        return
-    if isinstance(error, discord.ext.commands.CheckFailure):
-        await ctx.send("[!] I'm sorry " + ctx.author.name + ", I'm afraid I can't do that.")
-        return
-    if isinstance(error, discord.ext.commands.ExtensionError):
-        return  # Already handled in loader
-    if isinstance(error, discord.ext.commands.errors.BadArgument):
-        await ctx.send(f"[!] Invalid argument: {error.args[0]}")
-        return
-
-    print('Ignoring exception in command {}:'.format(ctx.command), file=sys.stderr)
-    traceback.print_exception(type(error), error, error.__traceback__, file=sys.stderr)
-
-bot.run(token)
+def run_bot():
+    bot = commands.Bot(command_prefix=PREFIX, description=DESCRIPTION, case_insensitive=True)
+    bot.version = BOT_VERSION
+    if len(modules_to_load) != 0:
+        for module in modules_to_load:
+            try:
+                bot.load_extension(module)
+                print(f"[LOAD] Successfully loaded extension {module}")
+            except Exception as e:
+                print(e)
+                print(f"[ERROR] Failed to load extension {module}, continuing.")
+    bot.run(token)
+
+
+if __name__ == '__main__':
+    while True:
```

Comment:
Is there a better way of implementing this? Not a big fan of a `while True` loop.

Reply:
After reading and asking around I really couldn't find a single discord.py who found this necessary. I don't think it's worth the extra trouble.